source/RemoteWiring/HardwareProfile.cpp

```diff
@@ -0,0 +1,381 @@
+
+                break;
+
+            case PinMode::ANALOG:
+
+                //set the mode enabled & store the analog resolution of this pin
+                ++i;
+                if( i < size )
+                {
+                    currentPinCapabilities |= static_cast<uint8_t>( PinCapability::ANALOG );
+                    analogResolutions->insert( std::make_pair( total_pins, data[i] ) );
+                }
+                else
+                {
+                    found_errors = true;    //we've failed to get all of the data
+                }
+                ++i;
+
+                //analog offset keeps track of the first pin found that supports analog read, tells us how many digital pins we have,
+                //and allows us to convert analog pins like "A0" to the correct pin number
+                if( analog_offset == 0xFF )
+                {
+                    analog_offset = total_pins;
+                }
+                ++num_analog_pins;
+
+                break;
+
+            case PinMode::PWM:
+
+                //set the mode enabled & store the pwm resolution of this pin
+                ++i;
+                if( i < size )
+                {
+                    currentPinCapabilities |= static_cast<uint8_t>( PinCapability::PWM );
+                    pwmResolutions->insert( std::make_pair( total_pins, data[i] ) );
+                }
+                else
+                {
+                    found_errors = true;    //we've failed to get all of the data
+                }
+                ++i;
+
+                break;
+
+            case PinMode::SERVO:
+
+                //set the mode enabled & store the servo resolution of this pin
+                ++i;
+                if( i < size )
+                {
+                    currentPinCapabilities |= static_cast<uint8_t>( PinCapability::SERVO );
+                    servoResolutions->insert( std::make_pair( total_pins, data[i] ) );
+                }
+                else
+                {
+                    found_errors = true;    //we've failed to get all of the data
+                }
+                ++i;
+
+                break;
+
+            case PinMode::I2C:
+
+                //the next bit determines if the mode is enabled, this should always be true
+                ++i;
+                if( i < size )
+                {
+                    if( data[i] == MODE_ENABLED )
+                    {
+                        currentPinCapabilities |= static_cast<uint8_t>( PinCapability::I2C );
+                    }
+                }
+                else
+                {
+                    found_errors = true;    //we've failed to get all of the data
+                }
+                ++i;
+
+                break;
+
+            default:
+                //this value isn't recognized. it is likely fatal, but also possible that new data was added to the query response. We will attempt to continue
+                ++i;
```

Comment:
yes, most current values will define a mode like ANALOG followed by a value which is either '1' for supported, a resolution, or other value. The idea is that if we don't understand a value, we try to skip it and continue. If it does follow this format, we're likely to skip the next value as well, but this tries to allow for forward compatibility as much as possible.